conbench/tests/api/test_benchmarks.py

```diff
@@ -422,6 +422,92 @@ def test_create_benchmark_after_run_was_created(self, client):
                     benchmark_result.run.hardware, attr
                 ) == int(value)
 
+    def test_create_benchmark_after_run_different_machine_info(self, client):
+        """
+        This test documents current behavior that I was initially only
+        suspecting and then wanted to confirm with code.
+
+        We can of course re-think the specification (expected behavior).
+
+        This is a special case in the API surface, but because it's API
+        behavior it's important that we think about specification first
+        (desired behavior), and then judge about the implementation.
+
+        I think this is an artifact of us allowing for potentially many
+        concurrent executors to submit results within a specific run without
+        requiring special coordination between those executors. That is, it is
+        expected that the submitted run metadata (like name and machine info)
+        are equivalent across all BenchmarkResult entities, and the fastest
+        submitter wins the price of DB insertion, while the other racers get
+        their info silently dropped.
```

Comment:
This is certainly one kind of behavior this might supply. Honestly, the history here as far as I remember was basically: make it as easy as possible to post a benchmarkResult (e.g. make the run, hardware, etc. entities for you, don't make you make those manually) but still be able to lump benchmarkResults into runs and not have a proliferation of hardwares (hardware types do the same thing as runs, where if you post details of them that are keys, it re-uses the entry in the database — but if it's different you'll create a new database), etc.

Then the question basically is: is that ease of posting worth the confusion | complication in the API?

Reply:
>is that ease of posting worth the confusion?

I tend to think so. For one thing, if we were to require creating a Run or Hardware before posting any results, it would be a breaking change that we'd have to fix in all CI systems. I think most CI systems are using this behavior precisely because it is easier.

Reply:
Love the discussion. I hoped to trigger this kind of thought exchange by making that current behavior a little more explicit. 

I have not picked a side yet. 

There certainly would be a better middle ground API-wise if we had a well-specified `RunCreate` schema that would be part of the `BenchmarkCreate` schema.

>  it would be a breaking change that we'd have to fix in all CI systems.

Happy to share my opinion about this again and again :) Given the current state of the project I want to encourage us to not be afraid of making brutal changes. It's fair to say that we are in a prototyping phase.

Reply:
>I want to encourage us to not be afraid of making brutal changes.

I completely agree! Having done a few of these breaking changes in the past year, though, I've learned the hard way that they take more thought and work than they may seem on the surface, and so when we prioritize features by value-for-effort we should always be conscious of that.